File: src/cpp/http/purge.cpp
#include "purge.h"

namespace fastly::http::purge {

void purge_surrogate_key(std::string_view surrogate_key) {
  return fastly::sys::http::purge::f_http_purge_purge_surrogate_key(
      static_cast<std::string>(surrogate_key));
}

void soft_purge_surrogate_key(std::string_view surrogate_key) {
  return fastly::sys::http::purge::f_http_purge_soft_purge_surrogate_key(
      static_cast<std::string>(surrogate_key));
}

} // namespace fastly::http::purge

File: src/http/purge.rs
use cxx::CxxString;

pub fn f_http_purge_purge_surrogate_key(surrogate_key: &CxxString) {
    fastly::http::purge::purge_surrogate_key(surrogate_key.to_str().expect("invalid utf8"))
        .expect("purge failed");
}

pub fn f_http_purge_soft_purge_surrogate_key(surrogate_key: &CxxString) {
    fastly::http::purge::soft_purge_surrogate_key(surrogate_key.to_str().expect("invalid utf8"))
        .expect("purge failed");
}
